ast/edit/option_editor.go

```diff
@@ -0,0 +1,153 @@
+package edit
+
+import (
+	"fmt"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+type EditFn func(opt *ast.OptionStatement) (ast.Expression, error)
+
+/*
+`Edit` passes the `OptionStatement` in the AST rooted at `node` that has the
+specified identifier to `editFn`.
+The function can have side effects on the option statement
+and/or return a non-nil `Expression` that is set as value for the option.
+If the value returned by the edit function is `nil` (or an error is returned) no new value is set
+for the option statement (but any, maybe partial, side effect is applied).
+`Edit` returns whether it could find and edit the option (possibly with errors) or not.
+*/
+func Edit(node ast.Node, optionIdentifier string, editFn EditFn) (bool, error) {
+	oe := &optionEditor{identifier: optionIdentifier, editFn: editFn, err: nil}
+	ast.Walk(oe, node)
+
+	if oe.err != nil {
+		return oe.found, oe.err
+	}
+
+	return oe.found, nil
+}
+
+// Creates an `EditFn` for setting the value of an `OptionStatement`.
+func EditValueFn(expr ast.Expression) EditFn {
+	return func(opt *ast.OptionStatement) (ast.Expression, error) {
+		return expr, nil
+	}
+}
+
+/*
+Creates an `EditFn` for updating the values of an `OptionStatement` that has an
+`ObjectExpression` as value.
+*/
+func EditObjectFn(keyMap map[string]values.Value) EditFn {
+	return func(opt *ast.OptionStatement) (ast.Expression, error) {
+		obj, ok := opt.Assignment.Init.(*ast.ObjectExpression)
+		if !ok {
+			return nil, fmt.Errorf("value is not an object expression, it is %s instead", opt.Assignment.Init.Type())
+		}
+
+		for _, p := range obj.Properties {
+			value, found := keyMap[p.Key.Name]
+			if found {
+				p.Value = GetLiteral(value)
+			}
+		}
+
+		return nil, nil
+	}
+}
+
+func GetLiteral(v values.Value) ast.Expression {
```

Comment:
Can you comment this function? Also I think it needs a better name maybe `CreateLiteral` or `ConstructLiteral`?